great_expectations/_get_context.py

```diff
@@ -0,0 +1,370 @@
+from __future__ import annotations
```

Comment:
Moved all of this logic to its own file since its been growing in complexity - don't think there's any harm in having it go through another import?

Reply:
Moving the file does seem to destroy the blame here. So it's hard to follow the logic changes.

Reply:
I ported everything over 1-to-1 but I did refactor `get_context` into three separate helper methods:
* `_get_cloud_context`
* `_get_file_context`
* `_get_ephemeral_context`

Just a bit of cleanup to make the method a bit more readable since it was growing unwieldy